src/redist/targets/GenerateLayout.targets

```diff
@@ -471,6 +471,36 @@
 
   </Target>
 
+  <Target Name="VerifyApphostTemplateIsIdentifcal" Condition=" '$(HostOSName)' == 'osx' "
+          AfterTargets="LayoutAppHostTemplate"
+          DependsOnTargets="DownloadBundledComponents">
+
+    <PropertyGroup>
+      <NETCoreAppHostpkgPath>$(DownloadsFolder)$(DownloadedNetCoreAppHostPackInstallerFileName)</NETCoreAppHostpkgPath> 
+      <NETCoreAppHostpkgExtractPath>$(IntermediateOutputPath)AppHostpkgExtract\</NETCoreAppHostpkgExtractPath>
+    </PropertyGroup>
+
+    <!-- The folder cannot exist for pkgutil to run -->
+    <RemoveDir Directories="$(NETCoreAppHostpkgExtractPath)" />
+
+    <Exec Command="pkgutil --expand-full '$(NETCoreAppHostpkgPath)' '$(NETCoreAppHostpkgExtractPath)'" />
+
+    <ItemGroup>
+      <AllFilesOfAppHostPkg Include="$(NETCoreAppHostpkgExtractPath)\**\*.*" />
+      <ApphostInPkg
+          Include="@(AllFilesOfAppHostPkg)"
+          Condition="'%(FileName)%(Extension)' == '$(AppHostExecutableName)'"/>
+    </ItemGroup>
+
+    <Exec Command="diff '@(NativeRestoredAppHostNETCore->Distinct())' '@(ApphostInPkg->Distinct())' " ContinueOnError='true'>
+      <Output TaskParameter="ExitCode" PropertyName="DiffErrorCode"/>
```

Comment:
Is this ever going to patch over multiple distinct items?  If it were, then I think it would be overwriting the exit code each time.

Reply:
no, it won't. But i'll add an guard for this too